syft/pkg/cataloger/java/parse_pom_xml.go

```diff
@@ -24,6 +24,10 @@ func parserPomXML(path string, content io.Reader) ([]*pkg.Package, []artifact.Re
 
 	var pkgs []*pkg.Package
 	for _, dep := range pom.Dependencies {
+		if strings.HasPrefix(dep.Version, "${") {
+			versionProperty := dep.Version[2 : len(dep.Version)-1]
+			dep.Version = pom.Properties.Entries[versionProperty]
+		}
```

Comment:
It might be useful to move this to a utility function, something like `resolveValue(pom, string) string`, which would do something like this and could be used to resolve every value when reading the pom -- e.g. a user _could_ use variables for `groupId` or `artifactId`. If, at some point in the future, we are able to resolve parent POMs, then this function would also be able to handle variable lookups in parents and the rest of the code would be unchanged. What do you think?

Also, I think this algorithm should probably be a little different -- the user _could_ have a variable as only a part of the string like `<version>1.2.${minor.version}-beta</version>` -- maybe using regex, something like (totally untested):

```go
var propertyMatcher = regexp.MustCompile("[$][{][^}]+[}]")

func resolveValue(pom gopom.Project, s string) string {
  return propertyMatcher.ReplaceAllStringFunc(s, func(match string) string {
    prop := match[:len(match)-1][2:]
    pom.Properties.Entries[prop]
  })
}
```

I could take a stab at this small refactoring, if you'd like.

Reply:
Great suggestions. This was my first cut at using `golang`

Reply:
I'll see if I can get to it before tomorrow